integration_tests/redis/redis.py

```diff
@@ -0,0 +1,115 @@
+from unittest import TestCase
+from unittest.mock import patch
+
+from feats.redis import FeatureStream
+from feats.redis import RedisClient
+from feats.redis import StreamIterator
+from feats.redis import NotConnectedError
+
+
+class RedisClientTests(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.client = RedisClient(host='redis')
+
+    def test_connection_property_raises_error(self):
+        self.client.disconnect()
+        with self.assertRaises(NotConnectedError):
+            self.client.connection
```

Comment:
I think the more interesting behavior is what happens when __getitem__ is called after a client is disconnected.